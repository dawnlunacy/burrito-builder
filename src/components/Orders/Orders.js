import React from 'react';
import { connect } from 'react-redux';
import './Orders.css';

const Orders = ({orders}) => {
  
console.log("ORDERS", orders)
  if (orders.length === 0) {
    return <p> No Orders Yet. </p>
  } else {
    const orderEls = orders.map(order => {
      return (
        <div className="order">
          <h3>{order.name}</h3>
          <ul className="ingredient-list">
            {order.ingredients.map(ingredient => {
              return <li>{ingredient}</li>
            })}
          </ul>
        </div>
      )
    });
  
  

  return (
    <section>
      { orderEls.length ? orderEls : <p>No orders yet!</p> }
    </section>
  )
}
}
export const mapStateToProps = state => ({
  orders: state.orders
})


export default connect(mapStateToProps, null)(Orders);