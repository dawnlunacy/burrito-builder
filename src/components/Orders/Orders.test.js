import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapStateToProps } from './Orders';

describe('Orders', () => {
  let wrapper;

  const mockOrders = [{
    id: 0,
    name: "Lacy",
    ingredients: ["AwesomeSauce", "Optimism"]
  }, 
  {
    id: 1,
    name: "Robbie",
    ingredients: ["Pants", "Pugs", "Sweaters"]
  }];

  beforeEach(() => {
    wrapper = shallow(<Orders
      key={Date.now()} 
      orders={mockOrders}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('it should return an object with orders', () => {

      const mockState = {
        orders: mockOrders
      };

      const expected = {
        orders: mockOrders
      }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    });
  });
});