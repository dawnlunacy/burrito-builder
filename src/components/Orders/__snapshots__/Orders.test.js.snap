// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    id={0}
    key="0"
  >
    <h3>
      Lacy
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="AwesomeSauce"
      >
        AwesomeSauce
      </li>
      <li
        key="Optimism"
      >
        Optimism
      </li>
    </ul>
  </div>
  <div
    className="order"
    id={1}
    key="1"
  >
    <h3>
      Robbie
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="Pants"
      >
        Pants
      </li>
      <li
        key="Pugs"
      >
        Pugs
      </li>
      <li
        key="Sweaters"
      >
        Sweaters
      </li>
    </ul>
  </div>
</section>
`;
